tests/test_network_discovery.py

```diff
@@ -0,0 +1,577 @@
+            "profile_172.16.0.1",
+            "uid9012",
+            "timewindow15",
+            "2023-04-01 14:00:00",
+            EvidenceType.ICMP_ADDRESS_MASK_SCAN,
+        ),
+    ],
+)
+def test_check_icmp_sweep_valid_scans(
+    mocker,
+    mock_db,
+    msg,
+    note,
+    profileid,
+    uid,
+    twid,
+    timestamp,
+    expected_evidence_type,
+):
+    network_discovery = ModuleFactory().create_network_discovery_obj(mock_db)
+
+    mock_set_evidence = mocker.patch.object(
+        network_discovery.db, "set_evidence"
+    )
+
+    network_discovery.check_icmp_sweep(
+        msg, note, profileid, uid, twid, timestamp
+    )
+
+    assert mock_set_evidence.call_count == 1
+
+    called_evidence = mock_set_evidence.call_args[0][0]
+    assert called_evidence.evidence_type == expected_evidence_type
+    assert called_evidence.attacker.value == profileid.split("_")[1]
+    assert called_evidence.profile.ip == profileid.split("_")[1]
+    assert called_evidence.timewindow.number == int(
+        twid.replace("timewindow", "")
+    )
+    assert called_evidence.uid == [uid]
+    assert called_evidence.timestamp == timestamp
+
+
+def test_check_icmp_sweep_unsupported_scan(mocker, mock_db):
+    network_discovery = ModuleFactory().create_network_discovery_obj(mock_db)
+
+    mock_set_evidence = mocker.patch.object(
+        network_discovery.db, "set_evidence"
+    )
+
+    network_discovery.check_icmp_sweep(
+        msg="Some other scan detected on 20 hosts",
+        note="OtherScan",
+        profileid="profile_192.168.0.1",
+        uid="uid3456",
+        twid="timewindow20",
+        timestamp="2023-04-01 15:00:00",
+    )
+
+    assert mock_set_evidence.call_count == 0
+
+
+@pytest.mark.parametrize(
+    "flow_info, existing_dhcp_flows, "
+    "expected_set_dhcp_flow_calls, "
+    "expected_get_dhcp_flows_calls",
+    [
+        # Testcase 1: First DHCP request in timewindow
+        (
+            {
+                "flow": {
+                    "requested_addr": "192.168.1.100",
+                    "uids": ["uid1234"],
+                    "starttime": "2023-04-01 12:00:00",
+                },
+                "profileid": "profile_192.168.1.2",
+                "twid": "timewindow5",
+            },
+            {},
+            1,
+            1,
+        ),
+        # Testcase 2: Multiple DHCP requests,
+        # but not enough to trigger evidence
+        (
+            {
+                "flow": {
+                    "requested_addr": "192.168.1.101",
+                    "uids": ["uid5678"],
+                    "starttime": "2023-04-01 12:01:00",
+                },
+                "profileid": "profile_192.168.1.2",
+                "twid": "timewindow5",
+            },
+            {"192.168.1.100": ["uid1234"]},
+            1,
+            2,
+        ),
+    ],
+)
+def test_check_dhcp_scan_no_evidence(
+    mocker,
+    mock_db,
+    flow_info,
+    existing_dhcp_flows,
+    expected_set_dhcp_flow_calls,
```

Comment:
hey @sekhar the format of existing_dhcp_flows used above is incorrect, it should be something like
{'192.168.39.110': ['CsECJm4tWaMr55wLSd', 'CBxcVy37AvaHXNj4H9'], '192.168.39.120': ['xyz', 'abcd']}
the keys are IPs, the values are a list of uids

Reply:
yes exactly, something like what you did in test_check_dhcp_scan_with_evidence()